### ISSUE TITLE
Menu : array_flip(): Can only flip string and integer

### ISSUE THREAD

**Issue by merijnponzo** [timber/timber#2576]
With Timber 2 , WPML and php8
I added multiple menu's but I got the message:
```
Warning: array_flip(): Can only flip string and integer values, entry skipped in //wordpress/websitename/vendor/timber/timber/lib/Factory/MenuFactory.php on line 186
```

My nav is outputting
```
array(4) {
  ["primary"]=>
  int(3)
  ["secondary"]=>
  NULL
  ["footer"]=>
  int(3)
  ["language"]=>
  NULL
}
```

So checking in MenuFactory.php on line 186 for NULL values would solve the problem
```
protected function get_menu_location(WP_Term $term) : ?string {
    // fix
    $safe_locations = get_nav_menu_locations();
    $safe_locations_ar = [];
    foreach($safe_locations as $key => $value){
        if($value){
            $safe_locations_ar[$key] = $value;
        }
    }
    $locations = array_flip($safe_locations_ar);
    return $locations[$term->term_id] ?? null;
}
```

**Comment by nlemoine** [comment]
@merijnponzo This is fixed, It will be available as soon the PR mentioned above is merged (soon hopefully).

**Comment by gchtr** [comment]
I’m closing this, because this was fixed in #2517. It will be available with then next 2.x release, though I can’t say when that will be exactly.
